EventFilter/HcalRawToDigi/plugins/PackerHelp.h

```diff
@@ -0,0 +1,480 @@
+     header |= ((fiber-1) & QIE8HeaderSpec::MASK_FIBER)<<QIE8HeaderSpec::OFFSET_FIBER;
+     header |= (capid0 & QIE8HeaderSpec::MASK_CAPID)<<QIE8HeaderSpec::OFFSET_CAPID;
+     header |= (fiberErr & QIE8HeaderSpec::MASK_FIBERERR)<<QIE8HeaderSpec::OFFSET_FIBERERR;
+     header |= (0x5 & QIE8HeaderSpec::MASK_FLAVOR)<<QIE8HeaderSpec::OFFSET_FLAVOER; //flavor
+     header |= (0x1 & QIE8HeaderSpec::MASK_HEADER_BIT)<<QIE8HeaderSpec::OFFSET_HEADER_BIT;
+
+     return header;
+  }
+
+  uint16_t packTPheader(const HcalTriggerPrimitiveSample &tpSample, int channelid){
+     uint16_t header =0;
+
+     header |= (channelid & TPHeaderSpec::MASK_CHANID)<<TPHeaderSpec::OFFSET_CHANID;
+     header |= (0x0 & TPHeaderSpec::MASK_RESV)<<TPHeaderSpec::OFFSET_RESV;
+     header |= (0 & TPHeaderSpec::MASK_TPERR)<<TPHeaderSpec::OFFSET_TPERR;
+     header |= (0x4 & TPHeaderSpec::MASK_FLAVOR)<<TPHeaderSpec::OFFSET_FLAVOER; //flavor
+     header |= (0x1 & TPHeaderSpec::MASK_HEADER_BIT)<<TPHeaderSpec::OFFSET_HEADER_BIT;
+
+     return header;
+  }
+
+  uhtrData* newUHTR( int uhtrIndex , int orn = 0 , int bcn = 0 , uint64_t evt = 0 ){
+    
+    // initialize vector of 16-bit words
+    uhtrs[uhtrIndex] = uhtrData(8);
+    // build header -- some information will be updated at the end    
+    
+    uint64_t presamples    = 10;
+    uint64_t uhtrCrate     = uhtrIndex&0xFF;
+    uint64_t uhtrSlot      = (uhtrIndex&0xF00)>>8; 
+// From Jeremy:
+// Set the firmware to zero, the firmware flavor to 0, the payload format to 1, and the event type to 1.
+    uint64_t fwFlavor      = 0;
+    uint64_t eventType     = 1;
+    uint64_t payloadFormat = 1;
+    uint64_t fwVersion     = 0;
+
+    uint64_t uhtrHeader1 = 0;
+    uhtrHeader1 |= (uint64_t(0x0)&MASK_DATA_LENGTH)<<OFFSET_DATA_LENGTH;
+    uhtrHeader1 |= (bcn&MASK_BCN)<<OFFSET_BCN;
+    uhtrHeader1 |= (evt&MASK_EVN)<<OFFSET_EVN;
+    uhtrHeader1 |= (uint64_t(0x0)&MASK_FILED_BY_AMC13)<<OFFSET_FILED_BY_AMC13;
+    
+    uint64_t uhtrHeader2 = 0;
+    uhtrHeader2 |= (uhtrCrate&MASK_CRATE_ID)<<OFFSET_CRATE_ID;
+    uhtrHeader2 |= (uhtrSlot&MASK_SLOT_ID)<<OFFSET_SLOT_ID;
+    uhtrHeader2 |= (presamples&MASK_PRESAMPLES)<<OFFSET_PRESAMPLES;
+    uhtrHeader2 |= (orn&MASK_ORN)<<OFFSET_ORN;
+    uhtrHeader2 |= (fwFlavor&MASK_FW_FLAVOR)<<OFFSET_FW_FLAVOR;
+    uhtrHeader2 |= (eventType&MASK_EVENT_TYPE)<<OFFSET_EVENT_TYPE;
+    uhtrHeader2 |= (payloadFormat&MASK_PAYLOAD_FORMAT)<<OFFSET_PAYLOAD_FORMAT;
+    uhtrHeader2 |= (fwVersion&MASK_FW_VERSION)<<OFFSET_FW_VERSION;
+
+    // push header into vector of 16-bit words
+    uhtrs[uhtrIndex][0] = (uhtrHeader1>>0)&0xFFFF ;
+    uhtrs[uhtrIndex][1] = (uhtrHeader1>>16)&0xFFFF ;
+    uhtrs[uhtrIndex][2] = (uhtrHeader1>>32)&0xFFFF ;
+    uhtrs[uhtrIndex][3] = (uhtrHeader1>>48)&0xFFFF ;
+    uhtrs[uhtrIndex][4] = (uhtrHeader2>>0)&0xFFFF ;
+    uhtrs[uhtrIndex][5] = (uhtrHeader2>>16)&0xFFFF ;
+    uhtrs[uhtrIndex][6] = (uhtrHeader2>>32)&0xFFFF ;
+    uhtrs[uhtrIndex][7] = (uhtrHeader2>>48)&0xFFFF ;
```

Comment:
this is not easy to generalize since we have uhtrs[uhtrIndex] goes from 0 to 7 while we change the 64bit header from uhtrHeader1 to uhtrHeader2. For previous one related to cdfHeader and so on, I've made a function to do so.